src/omnisharp/utils.ts

```diff
@@ -121,4 +133,46 @@ export async function debugTestStop(server: OmniSharpServer, request: protocol.V
 
 export async function isNetCoreProject(project: protocol.MSBuildProject) {
     return project.TargetFrameworks.find(tf => tf.ShortName.startsWith('netcoreapp') || tf.ShortName.startsWith('netstandard')) !== undefined;
+}
+
+function isBlazorWebAssemblyProject(project: MSBuildProject): boolean {
+    if (!project.IsExe) {
+        return false;
+    }
+
+    if (!isWebProject(project)) {
+        return false;
+    }
+
+    // So we're a EXE web projct. We now need to differentiate between a typical web app and a Blazor WASM
+    // project. WebApps are typically executables and netcoreapp targeting; therefore, we're going to make
+    // the assumption that any app that's an executable and targeting netstandard2.1 or great is Blazor WASM.
+
+    const netstandardTFM = findNetStandardTargetFramework(project);
+    if (netstandardTFM === undefined) {
+        return false;
+    }
+
+    const netstandardTFMVersionString = netstandardTFM.ShortName.substring('netstandard'.length);
+    const netstandardTFMVersion = parseFloat(netstandardTFMVersionString);
+    if (!netstandardTFMVersion || netstandardTFMVersion < 2.1) {
+        return false;
+    }
+
+    const projectDirectory = path.dirname(project.Path);
+    const razorFiles = glob.sync('**/*.razor', { cwd: projectDirectory });
+    if (!razorFiles || razorFiles.length === 0) {
+        // No Razor files in the project
+        return false;
+    }
+
+    return true;
+}
+
+function isWebProject(project: MSBuildProject): boolean {
+    let projectFileText = fs.readFileSync(project.Path, 'utf8');
+
+    // Assume that this is an MSBuild project. In that case, look for the 'Sdk="Microsoft.NET.Sdk.Web"' attribute.
+    // TODO: Have OmniSharp provide the list of SDKs used by a project and check that list instead.
+    return projectFileText.toLowerCase().indexOf('sdk="microsoft.net.sdk.web"') >= 0;
```

Comment:
we can expose this TODO stuff via the existing workspace information endpoint

Reply:
This was already there, just moved the method :smi